### PR TITLE
Add docker provider label for CAPI

### PR DESCRIPTION
**Pull request by benmoss**
Adds a label for the Docker provider for CAPI

@detiber @vincepri @fabriziopandini

### CODE REVIEW

File: label_sync/labels.yaml
Comment (detiber) [inline label_sync/labels.yaml:56]: I believe these labels are intended for k8s cloud-provider integration rather than cluster-api providers. It might be better to add it to the `cluster-api` specific configuration below to avoid syncing this to other repositories (such as k/k)

**Comment by cpanato** [comment]
@benmoss need to follow this https://github.com/kubernetes/test-infra/tree/master/label_sync#usage to make bazel happy

**Comment by benmoss** [comment]
@detiber @cpanato fixed both! thanks

**Review by cpanato — APPROVED** [review]
/lgtm

**Comment by vincepri** [comment]
Looking at the other labels, they're all generic and not specific to any provider. Is this something that we actually want?

**Comment by benmoss** [comment]
Not sure what you mean, we already have labels for every other provider: https://github.com/kubernetes-sigs/cluster-api/labels?q=area

**Comment by benmoss** [comment]
Ah, I get it now! The other providers are referring to the in-tree k8s cloud providers. I thought it was strange that we were putting all these CAPI providers in the general labels area but now I realize it's the k8s providers that are leaking into CAPI. Hmm, I guess I am open that maybe we don't need to categorize issues relating to CAPD? I just think it would be handy. Open to alternative label names too.

**Review by fabriziopandini — COMMENTED** [review]
/lgtm

**Comment by vincepri** [comment]
No strong opinions here, just wanted to point that out, I'll let others chime in

cc @CecileRobertMichon @fabriziopandini

**Comment by CecileRobertMichon** [comment]
No objections to adding a docker label from my side, especially since it's scoped to the CAPI repo and the docker provider is special since it lives in the core repo.

/lgtm

**Review by fejta — APPROVED** [review]
/hold

(anyone in cluster-api is free to unhold)

**Comment by detiber** [comment]
I don't see any comments that strongly object to this, so removing the hold
/hold cancel

**Pull request merged**
